src/simulate.jl

```diff
@@ -158,8 +158,8 @@ function simulate!(
     isempty(θ) || setθ!(m, θ)
 
     if length(β) ≠ length(m.feterm.piv)
-        padding = length(model.feterm.piv) - m.feterm.rank
-        append!(β, fill(-0.0, padding))
+        β = invpermute!(copyto!(fill(-0.0, length(m.feterm.piv)), β),
```

Comment:
Why reassign `β` to a new array rather than adding an `invpermute!` below the existing `append!`?